### ISSUE TITLE
Python hangs when server does not iterate over requests

### ISSUE THREAD

**Issue by kpayson64** [grpc/grpc#6891]
If a stream request server does not iterate over its requests, the server will hang.  See #6882 for an example.

**Comment by tseaver** [comment]
@nathanielmanistaatgoogle AFAICT, this fix is not yet released:  is that correct?  I am definitely seeing hang-on-exit behavior with `grpcio-0.14.0`.

**Comment by tseaver** [comment]
Oops, my (now deleted) comment should have been directed to #6857.

**Comment by nathanielmanistaatgoogle** [comment]
Punting from `1.1.0`; this will be nice to have whenever we can have it. For the moment this issue is still blocked on [issue 7032](https://github.com/grpc/grpc/issues/7032) and this issue will be trivial to address after the fix of [issue 7032](https://github.com/grpc/grpc/issues/7032).

**Comment by nathanielmanistaatgoogle** [comment]
This will always be high-priority deep down in my insides.

**Comment by nathanielmanistaatgoogle** [comment]
Reopening following [pull request 14765](https://github.com/grpc/grpc/pull/14765)'s revert of [pull request 14687](https://github.com/grpc/grpc/pull/14687).

**Comment by nathanielmanistaatgoogle** [comment]
Nope; this should stay open.
